### PR TITLE
Rn updated app version expression

### PR DESCRIPTION
**Pull request by Rohit25negi**
pull the app version from the upstream version which the source app version for the data source(Only for india env, because it is slower and can't work on ICDS scale)

ticket: https://dimagi-dev.atlassian.net/browse/QA-1025

### CODE REVIEW

File: custom/icds_reports/ucr/tests/test_daily_feeding_daily_Attendence.py
Comment (orangejenny) [inline custom/icds_reports/ucr/tests/test_daily_feeding_daily_Attendence.py]: nitpick: lowercase filename

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [inline custom/icds_reports/ucr/expressions.py]: `get_version_from_appverision_string` => `get_version_from_appversion_string`

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [inline custom/icds_reports/ucr/expressions.py]: For consistency's sake, use `is_linked_app` from `corehq.apps.app_manager.util`

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [inline custom/icds_reports/ucr/expressions.py]: `get_build_by_version` would likely be faster, since it doesn't return the entire app doc but does include `upstream_version`. Not sure if performance matters since this is india only.

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [inline custom/icds_reports/ucr/expressions.py]: If getting the app fails and it's a linked app, this version is likely incorrect. Would it be better to return a blank value instead, or to let this fail instead of capturing the error here?

**Review by orangejenny — COMMENTED** [review]
App manager code looks reasonable, made a few minor comments.

Not approving until @calellowitz also takes a look.

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: hey `get_build_by_version` requires the `version` to pass as argument. which we won't have in advance

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: Hey, My main motive for adding this is to allow the test cases to pass locally because UCR tests have the forms not necessarily app. 
And In case it fails to get the app from `app = get_app(domain_name, form.app_id)` it means the app does not exist (linked app even). In that case, I think it is safer to fall back to the original method which is to get the app version from the string in the form. Feel free to disagree, happy to discuss more.

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [reply custom/icds_reports/ucr/expressions.py]: Can't you pull it from the string as you're doing for regular apps?

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [reply custom/icds_reports/ucr/expressions.py]: That makes sense, thanks for explaining.

**Review by orangejenny — COMMENTED** [review]
Updates look good.

File: custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py
Comment (calellowitz) [inline custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py]: these should use the `override_settings` decorator to make sure the change is localized to the specific and there are downstream effects on the rest of the test suite

File: custom/icds_reports/ucr/expressions.py
Comment (calellowitz) [inline custom/icds_reports/ucr/expressions.py]: When does a 404 happen?

File: custom/icds_reports/ucr/expressions.py
Comment (calellowitz) [reply custom/icds_reports/ucr/expressions.py]: if this would be faster it is a good change to make, but won't block on it

**Review by calellowitz — COMMENTED** [review]
I think this is good to hotfix, but it would be good to fix the other things before merge.

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: explained here: https://github.com/dimagi/commcare-hq/pull/26889#discussion_r394399372

Comment (calellowitz) [reply custom/icds_reports/ucr/expressions.py]: but 404 means the app wasn't found? was mostly just confused because there is not a web request so curious what function was raising that

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: Oh yep. Great idea. I can do that. and I see its way faster.

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: Ow yeah.. it was get_app function.

**Comment by Rohit25negi** [comment]
@calellowitz / @orangejenny  I have made the requested changes.

**Review by calellowitz — APPROVED** [review]

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [reply custom/icds_reports/ucr/expressions.py]: It's a kind of weird quirk of `get_app`. There are app manager exceptions for `ModuleNotFound` and `FormNotFound` but not for when an application isn't found.

File: custom/icds_reports/ucr/expressions.py
Comment (orangejenny) [inline custom/icds_reports/ucr/expressions.py]: You still need to check if it's a linked app, `upstream_app` will be null for regular apps.

Since it's just the json from saved_build, not an app, just check that `doc_type` is `LinkedApplication`.

File: custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py
Comment (orangejenny) [inline custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py]: Shouldn't this be `icds`?

**Review by orangejenny — CHANGES_REQUESTED** [review]
Still need to check the app type.

File: custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py
Comment (calellowitz) [reply custom/icds_reports/ucr/tests/test_daily_feeding_daily_attendance.py]: `icds` is the real, although if the test is just testing "not india" this should be fine

File: custom/icds_reports/ucr/expressions.py
Comment (Rohit25negi) [reply custom/icds_reports/ucr/expressions.py]: Good Catch. Thanks

**Review by orangejenny — APPROVED** [review]

**Comment by Rohit25negi** [comment]
merging this.

File: custom/icds_reports/ucr/expressions.py
Comment (snopoke) [inline custom/icds_reports/ucr/expressions.py:315]: @rohit-dimagi should this be changed to `icds-staging` now?

Comment (snopoke) [reply custom/icds_reports/ucr/expressions.py:315]: @Rohit25negi ?

**Pull request merged**
